### DISCUSSION TITLE
MQTT Nodes & Functionality

### DISCUSSION THREAD

**Discussion by brainw0rm** [cables-gl/cables_docs#479] [Feature Requests]
Hello people!

I think it would be really fantastic if there was some love for the MQTT stuff in cables, since the few nodes that existed while back are no longer in the node browser and also flagged as outdated. 

Being able to combine cables with a MQTT broker like shiftr.io is really powerful and opens up so many exciting possibilities for projects that I think it should be considered a cornerstone for a creative tool like cables. Probably also needs a few more nodes/functionality besides updating the old ones.

Thank you for reading and for your fantastic work!

ahoi,
flamingo

**Comment by steam0r** [comment]
the mqtt ops in cables where already using shiftr.io or whatever broker you had, it's basically websockets and we think the same functionality should be easily patchable using the WebSocket op. are you missing something from that?

**Reply by brainw0rm** [reply]
Yes, I am using the old MQTT ops as for now, copy and pasting the outdated versions of them from old patches that are online since they are not in the ops browser anymore. With the websocket op it seems I can not connect to the shiftr.io broker for example. Unless I am doing something wrong. Also the websocket ops dont have any options to manage topic subscriptions with a broker? But I am not an expert on that topic. Just when trying to implement the functionality of MQTT brokers it seems cables is lacking in that department.

Maybe to clarify: At the moment we are building some applications that rely heavily on exchanging data via topics through a mqtt broker and we would love to use cables as an interface to process and monitor the streams of that data. But since the options seem very limited at this point in time we have to default to other environments a lot of the time.

Maybe it would be cool to just update and reimplement the old MQTT nodes into the op browser again since they still seem to work as intended?

ahoi!

**Comment by steam0r** [comment]
we are in steps of having "extensions" for cases like this, feel free to contact us on discord if you want to pick up on the old mqtt code and maintain it as an extension, would be awesome!

**Reply by brainw0rm** [reply]
Sounds exciting! I think I dont have the required programming skills to maintain the old mqtt stuff and provide support for them though ;). But lets see how it all turns out!
